server/src/main/java/org/opensearch/extensions/rest/RegisterRestActionsRequest.java

```diff
@@ -23,51 +24,48 @@
  * @opensearch.internal
  */
 public class RegisterRestActionsRequest extends TransportRequest {
-    private String uniqueId;
-    private List<String> restActions;
-    private List<String> deprecatedRestActions;
+    private final RegisterRestActions request;
 
     public RegisterRestActionsRequest(String uniqueId, List<String> restActions, List<String> deprecatedRestActions) {
-        this.uniqueId = uniqueId;
-        this.restActions = new ArrayList<>(restActions);
-        this.deprecatedRestActions = new ArrayList<>(deprecatedRestActions);
+        ExtensionIdentity identity = ExtensionIdentity.newBuilder().setUniqueId(uniqueId).build();
+        this.request = RegisterRestActions.newBuilder()
+            .setIdentity(identity)
+            .addAllRestActions(restActions)
+            .addAllDeprecatedRestActions(deprecatedRestActions)
+            .build();
```

Comment:
I haven't had time to follow up with a thorough design, but I think there may need to be one more bit of information passed from the extension here, and it would be nice to do it with this PR rather than a separate one.

It relates to [SDK #355](https://github.com/opensearch-project/opensearch-sdk-java/issues/355) and replaced/deprecated routes.  Basically:
 - we want the deprecated routes coming from the extensions to be verbatim (with `/_plugin` prefix) without making any changes on OpenSearch side (but with possible exception handling if the plugin is installed and the prefix is already taken, per the linked issue)
 - we want active/new routes coming from the extension to use the uniqueId and `/_extension` prefix  but also possibly register with a `/_plugin` prefix, per the linked issue.

So we may want to actually include some sort of "prefix" string.

This may be exactly what you are doing with ExtensionIdentity but I'm thinking perhaps that may need a second string argument.

Sorry this isn't fully thought out, but perhaps something we should all agree on (and possibly implement [SDK #355](https://github.com/opensearch-project/opensearch-sdk-java/issues/355) shortly after this).

Reply:
(To be clear, my suggestion is to add a second string field to Identity, so it would have a uniqueID and a prefix.)

Reply:
Had a chat with @dbwiddis, we'll add a new bool in a different PR to solve it. 
Will try to get this PR merged as the framework.